plugins/fcrdns

```diff
@@ -168,35 +167,35 @@ sub is_valid_localhost {
         $self->adjust_karma(1);
         $self->log(LOGDEBUG, "pass, is localhost");
         return 1;
-    };
+    }
 
     my $rh = $self->qp->connection->remote_host;
-    if ($rh && lc $self->qp->connection->remote_host eq 'localhost') {
-        $self->log(LOGDEBUG, "pass, remote_host is localhost");
-        return 1;
-    };
+    return 0 if ! $rh;
+    return 0 if lc $self->qp->connection->remote_host ne 'localhost';
 
     $self->adjust_karma(-1);
     $self->log(LOGINFO, "fail, not localhost");
```

Comment:
You are reading that wrong. We subtract karma from hosts that claim to be localhost (i.e., they have their rDNS set to 'localhost') but are not arriving from a loopback IP.